captcha/fields.py

```diff
@@ -39,24 +42,54 @@ def __init__(self, public_key=None, private_key=None, use_ssl=None,
             settings.RECAPTCHA_PRIVATE_KEY
         self.use_ssl = use_ssl if use_ssl is not None else getattr(
             settings, 'RECAPTCHA_USE_SSL', True)
+        self.wizard = wizard
 
         self.widget = ReCaptcha(
             public_key=public_key, use_ssl=self.use_ssl, attrs=attrs)
         self.required = True
+
         super(ReCaptchaField, self).__init__(*args, **kwargs)
 
-    def get_remote_ip(self):
+    def get_request_object(self):
         f = sys._getframe()
         while f:
             if 'request' in f.f_locals:
                 request = f.f_locals['request']
                 if request:
-                    remote_ip = request.META.get('REMOTE_ADDR', '')
-                    forwarded_ip = request.META.get('HTTP_X_FORWARDED_FOR', '')
-                    ip = remote_ip if not forwarded_ip else forwarded_ip
-                    return ip
+                    return request
             f = f.f_back
 
+    def get_remote_ip(self):
+        request = self.get_request_object()
+        if request:
+            remote_ip = request.META.get('REMOTE_ADDR', '')
+            forwarded_ip = request.META.get('HTTP_X_FORWARDED_FOR', '')
+            ip = remote_ip if not forwarded_ip else forwarded_ip
+            return ip
+
+    def get_cached_result(self):
+        request = self.get_request_object()
+        if request:
+            key = 'rcf-{path}cached-result'.format(path=request.path_info)
+
+            if key in request.session:
+                expires_time = request.session[key]
+                return expires_time > time.time()
+
+        return False
+
+    def set_cached_result(self):
+        request = self.get_request_object()
+        if request:
+            key = 'rcf-{path}cached-result'.format(path=request.path_info)
+            now = datetime.datetime.now()
```

Comment:
I'm worried about times without timezones. You should make all your times tz aware, but explicitly choose UTC as your zone. Makes for easy debugging. However, just create an issue for that. At worst the captcha is cached a little longer or shorter than expected if things go wrong.